src/full_node/blockchain.py

```diff
@@ -235,6 +235,7 @@ async def receive_block(
 
         # Always add the block to the database
         await self.block_store.add_block(block, sub_block)
+        self.sub_blocks[sub_block.header_hash] = sub_block
```

Comment:
This is not correct, we are adding the sub_block in reconsider peak already. The issue is that we are not adding the genesis block

Reply:
why are we adding to sub_blocks and blockstore in different places? aren't they supposed to be the same (we load block store into sub_blocks on init)?

Reply:
@mariano54